cirq/api/google/v2/program.proto

```diff
@@ -0,0 +1,142 @@
+syntax = "proto3";
+
+package cirq.api.google.v2;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "ProgramProto";
+option java_multiple_files = true;
+
+// A quantum program.
+message Program {
+  // The language in which the program is written.
+  Language language = 1;
+
+  // Programs can be specified by a circuit or a schedule.
+  oneof program {
+    // A circuit is an abstract representation as a series of moments, each
+    // moment having a set of gates that act on disjoint qubits. Circuits don't
+    // have absolute times for their operations (gates acting on qubits).
+    //
+    // When circuits are scheduled on hardware operations within a moment
+    // will all be scheduled to start at the same time, and the length of
+    // time associated with a moment will be equal to the duration of the
+    // longest operation in the moment. Subsequent moments will start as soon
+    // as the longest operation in the previous moment finishes.
+    Circuit circuit = 2;
+
+    // Schedules are a list of operations (gates acting on qubits) that specify
+    // absolute start times for the operations.
+    Schedule schedule = 3;
+  }
+}
+
+// The quantum circuit, specified as a series of moments (abstract
+// slices of times with gates acting on disjoint sets of qubits).
+message Circuit {
+  // The moments of the circuit, with the first element corresponding to the
+  // first set of operations to apply, etc.
+  repeated Moment moments = 1;
+}
+
+// A moment is a collection of operations that operate on a disjoint set of
+// qubits. Conceptually a moment represents operations that all occurring
+// the same finite period of time.
+message Moment {
+  // All of the operations in the moment. Each operation must act on different
+  // qubits.
+  repeated Operation operations = 1;
+}
+
+// The quantum circuit, specified as a series of operations at specific
+// start times.
+message Schedule {
+  // A list of all the operations and their absolute start times.
+  repeated ScheduledOperation scheduled_operations = 3;
+}
+
+// An operation occurring at a specific start time.
+message ScheduledOperation {
+  // Which operation is to be scheduled.
+  Operation operation = 1;
+
+  // The start time of the operation, with zero representing the absolute
+  // start of the circuit.
+  //
+  // This must be consistent with the moment structure and must be positive.
+  int64 start_time_picos = 2;
+}
+
+// The language in which the program is expressed.
+message Language {
+  // The name of the gate set being used.
+  string gate_set = 1;
```

Comment:
Is there a place someone can go to find out the list of valid gate_set values you can specify here?

Reply:
currently the answer is no.  But I will be future looking and put a location.